spec/webhook_spec.rb

```diff
@@ -2,90 +2,74 @@
 
 require 'spec_helper'
 
-describe EasyPost::Webhook do
-  let(:mock_webhook_no_id) { described_class.new }
+describe EasyPost::Services::Webhook do
+  let(:client) { EasyPost::Client.new(api_key: ENV['EASYPOST_TEST_API_KEY']) }
 
   describe '.create' do
     it 'creates a webhook' do
-      webhook = described_class.create(
+      webhook = client.webhook.create(
         url: Fixture.webhook_url,
       )
 
-      expect(webhook).to be_an_instance_of(described_class)
+      expect(webhook).to be_an_instance_of(EasyPost::Models::Webhook)
       expect(webhook.id).to match('hook_')
       expect(webhook.url).to eq(Fixture.webhook_url)
 
       # Remove the webhook once we have tested it so we don't pollute the account with test webhooks
-      webhook.delete
+      client.webhook.delete(webhook.id)
     end
   end
 
   describe '.retrieve' do
     it 'retrieves a webhook' do
-      webhook = described_class.create(
+      webhook = client.webhook.create(
         url: Fixture.webhook_url,
       )
 
-      retrieved_webhook = described_class.retrieve(webhook.id)
+      retrieved_webhook = client.webhook.retrieve(webhook.id)
 
-      expect(retrieved_webhook).to be_an_instance_of(described_class)
+      expect(retrieved_webhook).to be_an_instance_of(EasyPost::Models::Webhook)
       expect(retrieved_webhook.to_s).to eq(webhook.to_s)
 
       # Remove the webhook once we have tested it so we don't pollute the account with test webhooks
-      webhook.delete
+      client.webhook.delete(webhook.id)
     end
   end
 
   describe '.all' do
     it 'retrieves all webhooks' do
-      webhooks = described_class.all(
+      webhooks = client.webhook.all(
         page_size: Fixture.page_size,
       )
 
       webhooks_array = webhooks.webhooks
 
       expect(webhooks_array.count).to be <= Fixture.page_size
-      expect(webhooks_array).to all(be_an_instance_of(described_class))
+      expect(webhooks_array).to all(be_an_instance_of(EasyPost::Models::Webhook))
     end
   end
 
   describe '.update' do
     it 'updates a webhook' do
-      webhook = described_class.create(
+      webhook = client.webhook.create(
         url: Fixture.webhook_url,
       )
 
-      updated_webhook = webhook.update
+      updated_webhook = client.webhook.update(webhook.id)
 
-      expect(updated_webhook).to be_an_instance_of(described_class)
+      expect(updated_webhook).to be_an_instance_of(EasyPost::Models::Webhook)
 
-      webhook.delete # Remove the webhook once we have tested it so we don't pollute the account with test webhooks
-    end
-
-    it 'throws error for missing id' do
-      expect {
-        mock_webhook_no_id.update
-      }.to raise_error(EasyPost::Error)
```

Comment:
issue: don't remove tests

Reply:
This unit test no longer applies to the thread-safe version because there will be no API call made from an object, which we had to check previously: https://github.com/EasyPost/easypost-ruby/blob/778b1ca8f4cfdd02b4da0c30dd6965701927f28e/lib/easypost/webhook.rb#L22-L31